### PR TITLE
[FLOC-1057] Upgrade Docker on slave AMIs

### PR DESCRIPTION
**Pull request by exarkun**
Fixes https://clusterhq.atlassian.net/browse/FLOC-1057

With this change, Docker can be upgraded by re-generating AMIs.  The version installed by get.docker.com at the time the scripts are run will be installed on the AMI.

[Review on Reviewable](https://reviewable.io/reviews/clusterhq/build.clusterhq.com/133)

### CODE REVIEW

File: slave/ubuntu-14.04/cloud-init-base.sh [hunk 1]
```diff
@@ -42,7 +34,12 @@ apt-get install -y \
 	dpkg-dev \
 	enchant
 
-# Maybe also linux-source
+# Install whatever version is newest in the Docker repository.  If you wanted a
+# different version, you should have run the script at a different time.
+curl https://get.docker.com/ > /tmp/install-docker.sh
+sh /tmp/install-docker.sh
+
+# XXX Maybe also linux-source
```
Comment (adamtheturtle) [inline slave/ubuntu-14.04/cloud-init-base.sh]: :question: This isn't new but why would we want this?

File: slave/centos-7/cloud-init-base.sh [hunk 2]
```diff
@@ -39,6 +45,12 @@ yum install -y \
 	curl \
 	enchant
 
+# Install whatever version is newest in the Docker repository.  If you wanted a
+# different version, you should have run the script at a different time.
```
Comment (adamtheturtle) [inline slave/centos-7/cloud-init-base.sh]: What is important, I think, is that the way we install Docker here matches some docs somewhere about how users should install Docker. It is a shame that we are duplicating code, but for now add a comment (in both places, here and where the Docker install instructions are in the docs?) that this should match some particular docs.

(Same goes for Ubuntu)

**Comment by adamtheturtle** [comment]
:question: These AMIs are described in the README / Slaves / Slave AMIs section. That seems to be out of date. Now might be a good time to update that, as I (as a reviewer) looked there to see what these files were all about.

Thanks @exarkun , merge and deploy when you're happy. I haven't manually tested this but will leave that to you as you mentioned yesterday that you'd done some manual testing of this IIRC. In future (and sorry if I am misremembering) please describe what has been done in terms of manual testing. Then I can decide whether it is worth re-doing that / doing other manual testing.

**Comment by adamtheturtle** [comment]
Also, I decided to not care about pinning a version (as @tomprince suggested) because this now matches our install instructions. However it would be good to have some kind of process in place so that our software can know when Docker stable has changed and so we can do things (like upgrade the AMIs) as appropriate.

File: slave/ubuntu-14.04/cloud-init-base.sh (hunk 1 above)
Comment (exarkun) [inline slave/ubuntu-14.04/cloud-init-base.sh]: No idea.

**Pull request merged**